skyportal/tests/frontend/test_gcnevents.py

```diff
@@ -85,3 +91,123 @@ def test_gcnevents(
+            'elevation': 0.0,
+            'diameter': 10.0,
+        },
+        token=super_admin_token,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+    telescope_id = data['data']['id']
+
+    fielddatafile = f'{os.path.dirname(__file__)}/../../../data/ZTF_Fields.csv'
+    regionsdatafile = f'{os.path.dirname(__file__)}/../../../data/ZTF_Region.reg'
+
+    instrument_name = str(uuid.uuid4())
+    status, data = api(
+        'POST',
+        'instrument',
+        data={
+            'name': instrument_name,
+            'type': 'imager',
+            'band': 'Optical',
+            'filters': ['ztfr'],
+            'telescope_id': telescope_id,
+            'field_data': pd.read_csv(fielddatafile)[:5].to_dict(orient='list'),
+            'field_region': Regions.read(regionsdatafile).serialize(format='ds9'),
+        },
+        token=super_admin_token,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+
+    # wait for the fields to populate
+    time.sleep(15)
+
+    datafile = f'{os.path.dirname(__file__)}/../../../data/sample_observation_data.csv'
+    data = {
+        'telescopeName': name,
+        'instrumentName': instrument_name,
+        'observationData': pd.read_csv(datafile).to_dict(orient='list'),
+    }
+
+    status, data = api('POST', 'observation', data=data, token=super_admin_token)
+
+    assert status == 200
+    assert data['status'] == 'success'
+
+    # wait for the executed observations to populate
+    time.sleep(15)
+
+    driver.get(f'/become_user/{user.id}')
+    driver.get('/gcn_events/2019-04-25T08:18:05')
+
+    driver.wait_for_xpath('//*[text()="190425 08:18:05"]')
+    driver.wait_for_xpath('//*[text()="LVC"]')
+    driver.wait_for_xpath('//*[text()="BNS"]')
+
+    # test modify sources form
+    driver.wait_for_xpath('//*[@id="root_startDate"]').send_keys('04/24/2019')
+    driver.wait_for_xpath('//*[@id="root_startDate"]').send_keys(Keys.TAB)
+    driver.wait_for_xpath('//*[@id="root_startDate"]').send_keys('01:01')
+    driver.wait_for_xpath('//*[@id="root_startDate"]').send_keys('P')
+    driver.wait_for_xpath('//*[@id="root_endDate"]').send_keys('04/30/2019')
+    driver.wait_for_xpath('//*[@id="root_endDate"]').send_keys(Keys.TAB)
+    driver.wait_for_xpath('//*[@id="root_endDate"]').send_keys('01:01')
+    driver.wait_for_xpath('//*[@id="root_endDate"]').send_keys('P')
+    driver.wait_for_xpath('//*[@id="root_localizationCumprob"]').clear()
+    driver.wait_for_xpath('//*[@id="root_localizationCumprob"]').send_keys('1.01')
+    driver.wait_for_xpath('//*[@id="root_localizationName"]')
+    driver.click_xpath('//*[@id="root_localizationName"]')
+    driver.wait_for_xpath('//li[contains(text(), "bayestar.fits.gz")]')
+    driver.click_xpath('//li[contains(text(), "bayestar.fits.gz")]')
+
+    submit_button_xpath = '//button[@type="submit"]'
+    driver.wait_for_xpath(submit_button_xpath)
+    driver.click_xpath(submit_button_xpath)
```

Comment:
Shouldn't there be some test that when you click that last button it actually does what you want?

Reply:
OOps.